test/extensions/filters/http/common/factory_base_test.cc

```diff
@@ -0,0 +1,58 @@
+#include "envoy/server/filter_config.h"
+
+#include "common/protobuf/message_validator_impl.h"
+
+#include "extensions/filters/http/common/factory_base.h"
+
+#include "test/extensions/filters/http/common/factory_base_test.pb.validate.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/test_runtime.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace {
+
+class FakeFilterFactory
+    : public FactoryBase<test::extensions::filters::http::common::FakeFilterConfig> {
+public:
+  FakeFilterFactory() : FactoryBase("test") {}
+
+private:
+  virtual Http::FilterFactoryCb createFilterFactoryFromProtoTyped(
```

Comment:
nit: `virtual` isn't needed here, as the method has been declared as an `override`